src/ocean/core/SmartUnion.d

```diff
@@ -212,6 +212,86 @@ version (UnitTest)
     }
 }
 
+
+/*******************************************************************************
+
+    Calls the specified callable with the active field of the provided
+    smart-union. If no field is active, does nothing.
+
+    Note: declared at module-scope (rather than nested inside the SmartUnion
+    template) to work around limitations of template alias parameters. (Doing it
+    like this allows it to be called with a local name.)
+
+    Params:
+        Callable = alias for the thing to be called with the active member of
+            the provided smart-union
+        SU = type of smart-union to operate on
+        smart_union = smart-union instance whose active field should be passed
+            to Callable
+
+*******************************************************************************/
+
+public void callWithActive ( alias Callable, SU ) ( SU smart_union )
+{
+    alias TemplateInstanceArgs!(SmartUnion, SU) SUArgs;
+    static assert(SUArgs.length == 1);
+    alias SUArgs[0] U;
+
+    if ( smart_union._.active == smart_union.Active.none )
+        return;
+
+    assert(smart_union._.active >= 1);
```

Comment:
`smart_union.Active.none` is intentionally defined to be zero to allow for
```d
if ( !smart_union._.active ) return;
```
in situations like this, which makes the assertion unnecessary (at least for me ;) ).